scripts/core/superdesk-api.d.ts

```diff
@@ -35,12 +35,13 @@ declare module 'superdesk-api' {
 
     export interface IArticleAction {
         labelForGroup?: string;
+        groupId?: string; // action lists can specify which groups they wanna render via an id
```

Comment:
You shouldn't add groupId here since it won't work in other places `IArticleAction` is used. `labelForGroup` is used for the same purpose in monitoring item action. I added it expecting that a group label might actually be rendered which it isn't as of now. I suggest either removing `labelForGroup` or using another interface.

Since `priority` and `icon` aren't implemented either, you should definitely not use this interface. You can extend the interface and omit some properties to make the interface match the implementation:

```
interface IArticleEditViewAction extends Omit<IArticleAction, 'priority' | 'icon'> {
    groupId?: string; // action lists can specify which groups they wanna render via an id
}
```

Reply:
@tomaskikutis I'm thinking about refactoring [this](https://github.com/superdesk/superdesk-client-core/blob/249dc467858084540efa5b36ead70cc87ad4eb60/scripts/apps/search/components/actions-menu/MenuItems.tsx#L216-L228) to use an ID instead of the label

Reply:
Sure, go ahead.